realm/realm-library/src/objectServer/java/io/realm/SyncUser.java

```diff
@@ -203,7 +202,12 @@ private void postError(final ObjectServerError error) {
                     handler.post(new Runnable() {
                         @Override
                         public void run() {
-                            callback.onError(error);
+                            try {
```

Comment:
As noted in the issue, I actually think that throwing an exception in an async error handler is a really bad idea. For example RxJava does not allow it (or consider it bad practise). It might be difficult to catch an exception from an async task, so doing it might just terminate the app.